src/main/java/com/boomi/flow/services/boomi/mdh/database/FieldMapper.java

```diff
@@ -122,36 +128,129 @@
+            }
+
+            // if it is one of our properties we don't change the property name, but we still need to change the property names of the object in objectData
+            if (property.getContentType() == ContentType.List ||
+                    property.getContentType() == ContentType.Object || changePropertyName == false
+                    && property.getObjectData() != null) {
+
+                if (changePropertyName) {
+                    newName = universe.getLayout().getModel().getName() + " - " + newName;
+                }
+                for (MObject childObject:property.getObjectData()) {
+                    childObject.setDeveloperName(newName);
+                    renameMobjectPropertiesToUseUniqueId(universe, childObject);
+                }
+            }
+
+            property.setDeveloperName(newName);
+        }
+    }
+
+    /**
+     * when the response is translated automatically mapped to MObject the names are still name/collectionTag,
+     * we search in this method for a name that flow understand (uniqueId)
+     */
+    public static String getUniqueIdByPropertyName(String modelName, Property property, List<Universe.Layout.Model.Element> elements) {
+        String fieldName = property.getDeveloperName();
+
+        if (property.getContentType() == ContentType.List) {
+            fieldName = Entities.removeModelPrefix(property.getDeveloperName(), modelName);
+            Universe.Layout.Model.Element foundElement = findElementBy(elements, fieldName, SearchingBy.COLLECTION_TAG);
+
+            return foundElement.getUniqueId().toLowerCase();
+        } else if(property.getContentType() == ContentType.Object) {
+            fieldName = Entities.removeModelPrefix(property.getDeveloperName(), modelName);
+            Universe.Layout.Model.Element foundElement = findElementBy(elements, fieldName, SearchingBy.NAME);
+
             return foundElement.getUniqueId().toLowerCase();
-        } else {
-            return property.getDeveloperName();
         }
+
+        Universe.Layout.Model.Element foundElement = findElementBy(elements, fieldName, SearchingBy.NAME);
+
+        if (foundElement == null) {
+            throw new RuntimeException("The field "+ fieldName + " hasn't been found, please update the service");
+        }
+
+        return foundElement.getUniqueId().toLowerCase();
+    }
+
+    /*
+        this is a recursive search through all the element, we can search elements by collectionTag, Name or UniqueID
+     */
+    static Universe.Layout.Model.Element findElementBy(List<Universe.Layout.Model.Element> elements, String search, SearchingBy searchingBy) {
+        Universe.Layout.Model.Element found = null;
+        for(Universe.Layout.Model.Element element: elements) {
+            if (searchingBy == SearchingBy.NAME && element.getName().toLowerCase().equals(search) ||
```

Comment:
I'm guessing here you can use `equalsIgnoreCase`, and shouldn't `search` have its case ignored too?